br/pkg/lightning/restore/check_info.go

```diff
@@ -139,6 +177,108 @@ func (rc *Controller) ClusterIsAvailable(ctx context.Context) error {
 	return nil
 }
 
+func (rc *Controller) checkEmptyRegion(ctx context.Context) error {
+	passed := true
+	message := "Cluster doesn't have too many empty regions"
+	defer func() {
+		rc.checkTemplate.Collect(Critical, passed, message)
+	}()
+
+	var result api.RegionsInfo
+	if err := rc.tls.WithHost(rc.cfg.TiDB.PdAddr).GetJSON(ctx, pdEmptyRegions, &result); err != nil {
+		return errors.Trace(err)
+	}
+	regions := make(map[uint64]int)
+	for _, region := range result.Regions {
+		for _, peer := range region.Peers {
+			regions[peer.StoreId]++
+		}
+	}
+
+	var messages []string
+	for storeID, regionCnt := range regions {
+		if regionCnt > errorEmptyRegionCntPerStore {
+			passed = false
+			messages = append(messages, fmt.Sprintf("TiKV store %v contains too many empty regions, "+
```

Comment:
If there are a lot of regions that trigger this check, the error message will end up be too long. I think we should do some kind of deduplication

Reply:
It's a good suggestion.